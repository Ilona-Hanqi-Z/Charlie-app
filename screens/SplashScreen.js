import React from 'react';
import {
    View,
    Text,
    Button,
    Dimensions,
    StyleSheet,
    TouchableOpacity,
    Image
} from 'react-native';
import * as Animatable from 'react-native-animatable';
import {LinearGradient} from 'expo-linear-gradient';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

const SplahScreen = ({navigation}) => {
    return (
        <View style={styles.container}>
           
            <View style={styles.header}>
                <Animatable.Image 
                        animation="bounceIn"
                      
                    source={require('../assets/fresco.png')}
                    style={styles.logo}
                    reesizeMode="stretch"
                    />
            </View>
            <Animatable.View style={styles.footer}
                             animation="fadeInUpBig"
            >
                <Text style={styles.title}>Share Your Story With World!</Text>
                <Text style={styles.text}>Sign in with account</Text>
                <View style={styles.button}>
                <TouchableOpacity onPress = {() => navigation.navigate('SignInScreen')}>
                    <LinearGradient
                        colors={['#ffc600','#ffe900']}
                        style={styles.signIn}>
                        <Text style={styles.textSign}>Get Started</Text>
                        <MaterialIcons
                        name="navigate-next"
                        color="#fff"
                        size={20}
                        />
                    </LinearGradient>
                    
                </TouchableOpacity>
                </View>
                
            </Animatable.View>
        </View>
    );
};

export default SplahScreen;

const {height} = Dimensions.get("screen");
const height_logo = height*0.28;

const styles = StyleSheet.create({
    container: {
      flex: 1, 
      backgroundColor: '#ffc600'
    },
    header: {
        flex: 2,
        justifyContent: 'center',
        alignItems: 'center'
    },
    footer: {
        flex: 1,
        backgroundColor: '#fff',
        borderTopLeftRadius: 30,
        borderTopRightRadius: 30,
        paddingVertical: 50,
        paddingHorizontal: 30
    },
    logo: {
        width: height_logo,
        height: height_logo
    },
    title: {
        color: '#05375a',
        fontSize: 30,
        fontWeight: 'bold'
    },
    text: {
        color: 'grey',
        marginTop:5
    },
    button: {
        alignItems: 'flex-end',
        marginTop: 30
    },
    signIn: {
        width: 150,
        height: 40,
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 20,
        flexDirection: 'row'
    },
    textSign: {
        color: 'white',
        fontWeight: 'bold'
    }
  });