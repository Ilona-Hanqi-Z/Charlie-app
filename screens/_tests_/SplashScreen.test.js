import React from 'react';
import { render } from 'react-native-testing-library';
import SplashScreen from '../SplashScreen';

test('it renders all inputs as expected', ()=> {
    const { toJSON } = render(<SplashScreen />);

    expect(toJSON()).toMatchSnapshot();
})