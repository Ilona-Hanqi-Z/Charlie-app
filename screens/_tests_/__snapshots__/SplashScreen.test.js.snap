// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders all inputs as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffc600",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Image
      forwardedRef={[Function]}
      resizeMode="stretch"
      source={1}
      style={
        Object {
          "height": 373.52000000000004,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
          "width": 373.52000000000004,
        }
      }
    />
  </View>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "backgroundColor": "#f2f2ed",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 1,
        "opacity": 0,
        "paddingHorizontal": 30,
        "paddingVertical": 50,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Stay connected with everyone!
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "marginTop": 5,
        }
      }
    >
      Log In with account
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "marginTop": 30,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4280824123,
                4282930273,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 30,
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "center",
              "width": 150,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
            Get Started
          </Text>
          <Text />
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </View>
</View>
`;
